File: old-regressions/cs/expr.4.cs

/*++
Copyright (c) 2015 Microsoft Corporation

--*/

using Microsoft.Z3;
using System;
using System.Collections.Generic;

class Test
{
    public void Run()
    {
        Dictionary<string, string> cfg = new Dictionary<string, string>() { 
            { "AUTO_CONFIG", "true" } };

        using (Context ctx = new Context(cfg))
        {
            IntExpr x = ctx.MkIntConst("x");

            Console.WriteLine("is expression: " + (x is Expr));

            Expr n = ctx.MkAdd(x, ctx.MkInt(1));

            Console.WriteLine("is application: " + (x.IsNumeral || x.IsExpr));

            Console.WriteLine("decl: " + n.FuncDecl);

            Console.WriteLine("num args: " + n.NumArgs);

            for (uint i = 0; i < n.NumArgs; i++)
                Console.WriteLine("arg(" + i + ") -> " + n.Args[i]);
        }
    }
}

File: old-regressions/python/expr.4.py

# Copyright (c) 2015 Microsoft Corporation
from z3 import *
set_option(auto_config=True)

x = Int('x')
print "is expression: ", is_expr(x)
n = x + 1
print "is application:", is_app(n)
print "decl:          ", n.decl()
print "num args:      ", n.num_args()
for i in range(n.num_args()):
    print "arg(", i, ") ->", n.arg(i)
